### PR TITLE
Amazon onebox improvements

### PR DESCRIPTION
**Pull request by majakomel**
Meta: https://meta.discourse.org/t/import-data-from-e-g-amazon-like-isbn-etc-a-extended-onebox-solution-for-amazon/79565

### CODE REVIEW

**Comment by majakomel** [comment]
Rubocop is very unhappy with the safe navigation operator. Is it ok to update `TargetRubyVersion` to 2.3 for rubocop?

**Comment by SamSaffron** [comment]
2.2 is EOL in another 3 months so I guess yes, its fine to drop 2.2 now. We
just need to document this in the Readme and make sure the gem is explicit
about it.

File: lib/onebox/engine/amazon_onebox.rb [hunk 1]
```diff
@@ -102,12 +102,12 @@ def data
             by_info: authors,
             image: og[:image] || image,
             description: raw.at("#productDescription")&.inner_text,
-            rating: rating,
+            rating: "#{rating}%s" % (', ' if rating && (!isbn&.empty? || !price&.empty?)),
```
Comment (glebm) [inline lib/onebox/engine/amazon_onebox.rb]: You don't need a format string here, can just do:

```ruby
rating: "#{rating}#{', ' if rating && (!isbn&.empty? || !price&.empty?)}"
```

**Comment by SamSaffron** [comment]
cool, lets try this out, thanks @majakomel

**Comment by SamSaffron** [comment]
long term I am a bit uncomfortable with all these enormous html fixtures living in the repo cause this is bloating the size of the gemfile and traditionally you are meant to ship gems with specs which we do... but our spec dir is getting big, it is 6.4mb uncompressed.

**Comment by glebm** [comment]
> traditionally you are meant to ship gems with specs which we do

That doesn't seem right. Bundler [excludes tests](https://github.com/bundler/bundler/blob/86e4b2a636fa2153bcd048dc084ace180cfe5c4a/lib/bundler/templates/newgem/newgem.gemspec.tt#L32) when you create a new gem.

**Comment by SamSaffron** [comment]
yes I just noticed that, so I excluded all specs here as well.

**Pull request merged**